File: db4oj/core/src/com/db4o/MSetSemaphore.java
/* Copyright (C) 2004   db4objects Inc.   http://www.db4o.com */

package com.db4o;

import com.db4o.foundation.network.YapSocket;

final class MSetSemaphore extends MsgD {
	final boolean processMessageAtServer(YapSocket sock) {
		int timeout = readInt();
		String name = readString();
		YapFile stream = (YapFile)getStream();
		boolean res = stream.setSemaphore(getTransaction(), name, timeout);
		if(res){
			Msg.SUCCESS.write(stream, sock);
		}else{
			Msg.FAILED.write(stream, sock);
		}
		return true;
	}
}

File: db4on/core/src/core/com/db4o/cs/messages/MSetSemaphore.cs
namespace com.db4o.cs.messages
{
	public sealed class MSetSemaphore : com.db4o.cs.messages.MsgD
	{
		public sealed override bool ProcessMessageAtServer(com.db4o.foundation.network.YapSocket
			 sock)
		{
			int timeout = ReadInt();
			string name = ReadString();
			com.db4o.YapFile stream = (com.db4o.YapFile)GetStream();
			bool res = stream.SetSemaphore(GetTransaction(), name, timeout);
			if (res)
			{
				com.db4o.cs.messages.Msg.SUCCESS.Write(stream, sock);
			}
			else
			{
				com.db4o.cs.messages.Msg.FAILED.Write(stream, sock);
			}
			return true;
		}
	}
}
